### PR TITLE
Task/HW-57734 Update Graph QL to retrieve field formatting rules

### PR DESCRIPTION
**Pull request by pasohan-altmtrk**
Update Graph QL to retrieve field formatting rules Initial Version has drafted

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public List<ConditionalPattern> getConditionalPatterns() {
+        return mConditionalPatterns;
+    }
+
+    public boolean containsConditionalPattern() {
+        return mConditionalPatterns != null && !mConditionalPatterns.isEmpty();
+    }
+
+    public String getConditionalPattern(@NonNull final String value) {
+        for (ConditionalPattern pattern : mConditionalPatterns) {
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java]: I have added the containsConditionalPattern() check on before conditional pattern loop

File: core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    public List<ConditionalPattern> getConditionalPatterns() {
+        return mConditionalPatterns;
+    }
+
+    public boolean containsConditionalPattern() {
+        return mConditionalPatterns != null && !mConditionalPatterns.isEmpty();
+    }
+
+    public String getConditionalPattern(@NonNull final String value) {
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java]: Rename has been done

File: core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                if (conditionalPatternJson != null) {
+                    mConditionalPatterns.add(new ConditionalPattern(conditionalPatternJson));
+                }
+            }
+        } else {
+            mConditionalPatterns = null;
+        }
+    }
+
+    public String getDefaultPattern() {
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java]: Agree, Both cases covered in getPattern method itself

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/HyperwalletFieldTest.java [hunk 4]
```diff
@@ -63,4 +63,26 @@ public void testHyperwalletField_convertJsonObject() throws JSONException {
         assertThat(hyperwalletField.getFieldSelectionOptions(), is(hasSize(2)));
         assertThat(hyperwalletField.getHyperwalletValidationMessage(), is(notNullValue()));
     }
+
+    @Test
+    public void testHyperwalletField_convertJsonObjectWithoutConditionalFormat() throws JSONException {
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/HyperwalletFieldTest.java]: Rename has been done

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/HyperwalletFieldTest.java [hunk 5]
```diff
@@ -63,4 +63,26 @@ public void testHyperwalletField_convertJsonObject() throws JSONException {
...
+                "mask_without_conditional_formatting_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+
+        assertTrue(hyperwalletField.containsMask());
+        assertThat(hyperwalletField.getMask(), is(notNullValue()));
+    }
+
+    @Test
+    public void testHyperwalletField_convertJsonObjectWithConditionalFormat() throws JSONException {
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/HyperwalletFieldTest.java]: Rename has been done

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+        Mask mask = hyperwalletField.getMask();
+
+        assertThat(mask.getDefaultPattern(), is("#### #### #### ####"));
+        assertThat(mask.getScrubRegex(), is("\\s"));
+        assertTrue(mask.containsConditionalPattern());
+        assertThat(mask.getConditionalPatterns(), is(notNullValue()));
+        assertThat(mask.getConditionalPatterns().size(), is(2));
+        assertThat(mask.getConditionalPattern("653"), is("######## ####### ####"));
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: Both Cases like

-  where a given value does not match any conditional pattern and 
- where a given value matches a conditional pattern
covered

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test
+    public void testGetConditionalPattern_returnsDefaultPattern() throws JSONException {
+        String jsonResponse = mExternalResourceManager.getResourceContent(
+                "mask_with_conditional_formatting_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+        Mask mask = hyperwalletField.getMask();
+
+        assertNotNull(mask);
```
Comment (dnip-hw) [inline core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: use assertThat

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test
+    public void testGetConditionalPattern_returnsPattern() throws JSONException {
+        String jsonResponse = mExternalResourceManager.getResourceContent(
+                "mask_with_conditional_formatting_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+        Mask mask = hyperwalletField.getMask();
+
+        assertNotNull(mask);
```
Comment (dnip-hw) [inline core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: use assertThat

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test
+    public void testHyperwalletFieldMask_convertJsonObjectWithConditionalPattern() throws JSONException {
+        String jsonResponse = mExternalResourceManager.getResourceContent(
+                "mask_with_conditional_formatting_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+        Mask mask = hyperwalletField.getMask();
+
+        assertNotNull(mask);
```
Comment (dnip-hw) [inline core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: use assertThat

File: core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
+/*
+ *  The MIT License (MIT)
+ *  Copyright (c) 2018 Hyperwallet Systems Inc.
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/main/java/com/hyperwallet/android/model/graphql/field/Mask.java]: Rectified and Pushed

File: core/src/main/java/com/hyperwallet/android/model/graphql/field/ConditionalPattern.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
+/*
+ *  The MIT License (MIT)
+ *  Copyright (c) 2018 Hyperwallet Systems Inc.
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/main/java/com/hyperwallet/android/model/graphql/field/ConditionalPattern.java]: Rectified and Pushed

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    public void testHyperwalletFieldMask_convertJsonObjectWithoutConditionalPattern() throws JSONException {
+        String jsonResponse = mExternalResourceManager.getResourceContent(
+                "mask_without_conditional_formatting_response.json");
+        JSONObject jsonResponseObject = new JSONObject(jsonResponse);
+        HyperwalletField hyperwalletField = new HyperwalletField(jsonResponseObject);
+        Mask mask = hyperwalletField.getMask();
+
+        assertNotNull(mask);
+        assertThat(mask.getScrubRegex(), is("\\s"));
+        assertNull(mask.getConditionalPatterns());
```
[the remark this answers is not in the document]
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: getPattern I have covered the two cases that will return

- default pattern when condition not matches
- conditional pattern when condition matches

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java (hunk 8 above)
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: Changes have been done

File: core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java (hunk 9 above)
Comment (pasohan-altmtrk) [reply core/src/test/java/com/hyperwallet/android/model/graphql/field/MaskTest.java]: Changes have been done

**Pull request merged**